tests/ip_link.py

```diff
@@ -17,14 +17,15 @@ def runTest(self):
         board.expect(prompt)
         if "ip: not found" not in board.before:
             up_interfaces = re.findall('\d: ([A-Za-z0-9-\.]+)[:@].*state UP ', board.before)
+            ether_interfaces = re.findall('\d: ([A-Za-z0-9-\.]+)[:@].*\n.*link/ether ', board.before)
+            assert up_interfaces == ether_interfaces
```

Comment:
We want to assume that all ethernet interfaces are up, right? In this case if any interface changes its state from UNKNOWN to UP, ane ethernet goes down, the amount of up interfaces will match the amount of ethernet interfaces. It'd be better to just for all interface types grep for the proper state and make sure they all match what we are expecting.